src/S3/S3ClientTrait.php

```diff
@@ -213,15 +216,38 @@ public function determineBucketRegionAsync($bucketName)
         return $handler($command)
             ->then(static function (ResultInterface $result) {
                 return $result['@metadata']['headers']['x-amz-bucket-region'];
-            }, static function (AwsException $exception) {
-                $response = $exception->getResponse();
+            }, function (AwsException $e) {
+                $response = $e->getResponse();
                 if ($response === null) {
-                    throw $exception;
+                    throw $e;
+                }
+
+                if ($e->getAwsErrorCode() === 'AuthorizationHeaderMalformed') {
```

Comment:
Just a bit thought, I'm open on this. Could we make the error code check at `determineBucketRegionFromExceptionBody` function instead perhaps? This helps reduce duplicate code, and easy to extend if there will be another special case in the future. 
Hmm, yet it might be a bit trickier to do this at the Multi region client.